src/tests/internal/test_azure.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2024 Recursive AI
+
+import os
+from unittest.mock import AsyncMock, Mock
+
+import pytest
+
+from recursiveai.benchmark._internal._llm._llm_model import ChatMessage
+from recursiveai.benchmark._internal._llm._azure_openai_gpt_model import (
+    AZURE_GPT,
+    AzureGPTX,
+)
+
+
+@pytest.fixture(autouse=True, scope="module")
+def skip_if_api_key_not_defined():
+    if os.getenv("OPENAI_API_KEY") is None:
+        pytest.skip("Skipping Azure OpenAI test since OPENAI_API_KEY is not defined")
+    if os.getenv("AZURE_OPENAI_DEPLOYMENT") is None:
+        pytest.skip("Skipping Azure OpenAI test since AZURE_OPENAI_DEPLOYMENT is not defined")
```

Comment:
I think that, besides AZURE_OPENAI_DEPLOYMENT, we also need to make sure that the env variable AZURE_OPENAI_ENDPOINT is set, otherwise it'll probably not work

Reply:
Added.